synapse/handlers/oidc_handler.py

```diff
@@ -869,6 +869,9 @@ async def _map_userinfo_to_user(
             raise MappingException(
                 "Failed to extract subject from OIDC response: %s" % (e,)
             )
+        # Some OIDC providers use integer IDs, but Synapse expects them to be
+        # strings. Really make sure it is a string.
+        remote_user_id = str(remote_user_id)
```

Comment:
won't this still cause problems due to numeric userids being reserved for guests?

Reply:
I do not believe so since the `map_user_attributes` call below returns what will be used for the localpart of the mxid.

The GitHub example given uses a string there (I think GItHub has an immutable integer ID for each user and then the string ID, e.g. clokep for me).

Pretty much this `remote_user_id` is only put into the `user_external_ids` table. I can add a comment to clarify that though.

Reply:
ohh I see. Ok!

Reply:
I clarified this slightly above (that this is used for the external ID).